rates/private/rates_support.f90

```diff
@@ -282,6 +280,20 @@ subroutine do_make_rate_tables( &
                      write(*, '(a,i4,2x,a)') 'missing raw rate for ',  &
                         j, trim(reaction_Name(reaction_id(j)))
                      a_okay = .false.
+                  else
+                    ! **New Addition: Store the logarithm of the rate**
+                      if (rattab(j, i) > 0.0_dp) then  ! Only take log of positive values
+                          rate_logR = log10(rattab(j, i))
+                      else if (rattab(j, i) == 0.0_dp) then
+                          rate_logR = -999d0
+                      else
+                          ! warning for rates set to -1
+                          ! like rni56ec_to_co56,rco56ec_to_fe56
+                          !write(*, '(a,i4,2x,a)') 'Warning: non-positive value for rate in ',  &
+                          !   j, trim(reaction_Name(reaction_id(j)))
+                          rate_logR = rattab(j, i) ! don't convert to log space.
```

Comment:
I could be misunderstanding what is going on here. Are you saying the data tables (in linear space) have some -1 entries? Then setting rate_logR = -1 could be a problem, no?